NetUtils.py

```diff
@@ -303,14 +304,21 @@ class Hint(typing.NamedTuple):
     found: bool
     entrance: str = ""
     item_flags: int = 0
+    priority: bool = True
 
     def re_check(self, ctx, team) -> Hint:
         if self.found:
             return self
         found = self.location in ctx.location_checks[team, self.finding_player]
         if found:
             return Hint(self.receiving_player, self.finding_player, self.location, self.item, found, self.entrance,
-                        self.item_flags)
+                        self.item_flags, self.priority)
+        return self
+    
+    def re_prioritize(self, ctx, priority: bool) -> Hint:
+        if priority != self.priority:
+            return Hint(self.receiving_player, self.finding_player, self.location, self.item, self.found, self.entrance,
+                        self.item_flags, priority)
```

Comment:
NamedTuple supports a `_replace` method, which return a new instance with fields replaced. 

```suggestion
            return self._replace(priority=priority)
```

(could also be applied to `re_check`)

Note that those methods are prefixed by an underscore, not because their usage is forbid, but to prevent conflict with fields (as per the documentation).